### ISSUE TITLE
Cloudant doesn't handle invalid URL properly

### ISSUE THREAD

**Issue by jannyHou** [cloudant/nodejs-cloudant#167]
In file `lib/reconfigure.js` it doesn't check whether the `url` parsed properly or not. 

With an invalid `url` string like "invalid_url",  [`parsed.hostname.match`](https://github.com/cloudant/nodejs-cloudant/blob/master/lib/reconfigure.js#L23) calls function `match()` against a `null` value(parsed.hostname). 
It is not a caught error that passed into callback, and app crashes with an error stack instead of connection error message:

```shell
if (parsed.hostname.match(/cloudant\.com$/) && parsed.protocol == "http:") {
                       ^

TypeError: Cannot read property 'match' of null
    at module.exports (/Users/jannyhou/lb/interrupt/cloudant-url-error-msg/node_modules/cloudant/lib/reconfigure.js:23:24)
    at Cloudant (/Users/jannyhou/lb/interrupt/cloudant-url-error-msg/node_modules/cloudant/cloudant.js:49:14)
    at Object.<anonymous> (/Users/jannyhou/lb/interrupt/cloudant-url-error-msg/cloudant-handle-bad-url/test.js:13:1)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:139:18)
    at node.js:968:3

```

I created a script to reproduce the error:
https://github.com/jannyHou/cloudant-handle-bad-url

There are two tests in `test.js`
  - First one catch the connection error properly 
  - Second one crashes with error stack.

**Comment by glynnbird** [comment]
@jannyHou looking at this now. Thanks for raising the issue.

**Comment by glynnbird** [comment]
Fixed in v1.7.0
